### ISSUE TITLE
Error "Symbol's function definition is void" after upgrading to 31519d393a

### ISSUE THREAD

**Issue by evanleck** [doomemacs/doomemacs#6462]
### What did you expect to happen?

Running `doom install` or `doom upgrade` should work. 

### What actually happened?

Running `doom install` or `doom upgrade` after upgrading to commit `31519d393a` throws an error.

```
x There was an unexpected error
  Message: Symbol's function definition is void
  Error: (void-function doplist!)
  Backtrace:
    (doplist! ((prop val) (list :pin "17cfa1b54800fdef2975c0c0531dad34846a5065...
    (let* ((name 'auto-minor-mode) (plist (cdr (assq name doom-packages)))) (l...
    (load-with-code-conversion "/Users/evan/.emacs.d/core/packages.el" "/Users...
    (load "/Users/evan/.emacs.d/core/packages.el" noerror nomessage nosuffix)
    (if (not noeval) (load file noerror 'nomessage 'nosuffix) (if (file-exists...
    (progn (if (not noeval) (load file noerror 'nomessage 'nosuffix) (if (file...
    (unwind-protect (progn (if (not noeval) (load file noerror 'nomessage 'nos...
    (save-current-buffer (set-buffer temp-buffer) (unwind-protect (progn (if (...
    (let ((temp-buffer (generate-new-buffer " *temp*" t))) (save-current-buffe...
    (condition-case e (let ((temp-buffer (generate-new-buffer " *temp*" t))) (...
! Extended backtrace logged to .emacs.d/.local/doom.error.log
x There was an unexpected error
  Message: Error with packages
  Error: (doom-package-error (:core . packages) "/Users/evan/.emacs.d/core/packages.el" (no-catch exit 255))
  Backtrace:
    (signal doom-package-error ((:core . packages) "/Users/evan/.emacs.d/core/...
    (condition-case e (let ((temp-buffer (generate-new-buffer " *temp*" t))) (...
    (doom--read-packages "/Users/evan/.emacs.d/core/packages.el" nil noerror)
    (let ((packages-file (concat doom-packages-file ".el")) doom-disabled-pack...
    (doom-package-list nil core)
    (let ((packages (doom-package-list nil 'core))) (let* ((--cl-rest-- (alist...
    (progn (if doom-debug-p (progn (let ((inhibit-message t)) (message #("DOOM...
    (if (or force-p (null (and (boundp 'straight-recipe-repositories) straight...
    (doom-initialize-core-packages)
    (load-with-code-conversion "/Users/evan/.emacs.d/core/core-cli.el" "/Users...
! Extended backtrace logged to .emacs.d/.local/doom.error.log
No catch for tag: exit, 255
```

### Describe your attempts to resolve the issue

I tried removing `~/.emacs.d/.local` and reinstalling, but to no avail.

### Steps to reproduce

Upgrade your local copy of doom to commit `31519d393a` and run `doom install` or `doom upgrade`.

### System Information

https://pastebin.com/UTMqw5JW

**Comment by zeroDivisible** [comment]
Confirming the issue is happening for me too.

**Comment by evanleck** [comment]
This is almost certainly caused by cb03d3258d05026796c4217bece9d7c01c659e8e.

**Comment by mrdev023** [comment]
Same for me

**Comment by mayersj1** [comment]
Confirming same issue here on MacOS 12.4. Tried deleting .emacs.d, cloning the repository again to no avail. This error occurs when running ~/.emacs.d/bin/doom install

**Comment by mrdev023** [comment]
Fixed here c4ac2ab384228f8c13eae641d7288739305af036

**Comment by mayersj1** [comment]
Pulled lastest and doom install is now running.  Will update once completed.

**Comment by mayersj1** [comment]
Seems to be stuck on 'Cloning evil-markdown'.. Any thoughts?

**Comment by evanleck** [comment]
I think @mrdev023 is right that this is fixed, I pulled the latest and am no longer getting an error.

**Comment by hlissner** [comment]
My bad! This regression was introduced in cb03d3258d05026796c4217bece9d7c01c659e8e
 and was fixed in c4ac2ab384228f8c13eae641d7288739305af036. If you're unable to update with `doom upgrade` because of the error, here's how to upgrade manually:

```sh
cd ~/.emacs.d
git pull
doom sync -u
```
